### PR TITLE
Removed duplicate in SpecificPriceRule::getAffectedProducts

### LINKED ISSUE #28062: PHP8.0 - an exception is displayed when creating a catalog price rule with a specific condition
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Using PHP8.0, when we create a catalog price rule with a specific condition
An exception is displayed
image

### Expected behavior

_No response_

### Steps to reproduce

1. Go to BO > Catalog > Discounts > Catalog price rule page
2. Create a catalog price rule with any discount
3. In the condition section, add Condition group 1 for the men category and add  Condition group 2 for the clothes category
image
4. Save 
5. see error 

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.0

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Removed duplicate in SpecificPriceRule::getAffectedProducts
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28062
| How to test?      | Cf #28062

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28104)

### CODE REVIEW

File: classes/SpecificPriceRule.php [hunk 1]
```diff
@@ -144,6 +144,8 @@ public function apply($products = false)
 
         $this->resetApplication($products);
         $products = $this->getAffectedProducts($products);
+        dump($this->id);
+        dump($products);
```
Comment (NeOMakinG) [inline classes/SpecificPriceRule.php]:
```suggestion
```

**Review by NeOMakinG — DISMISSED** [review]

File: classes/SpecificPriceRule.php [hunk 2]
```diff
@@ -275,8 +275,11 @@ public function getAffectedProducts($products = false)
                     $query->select('NULL as `id_product_attribute`');
                 }
 
+                // Merge previous result to current results
                 $result = array_merge($result, Db::getInstance()->executeS($query));
             }
+            // Remove duplicate after the array_merge
+            $result = array_intersect_key($result, array_unique(array_map('serialize', $result)));
```
Comment (PierreRambaud) [inline classes/SpecificPriceRule.php]:
```suggestion
            $result = array_unique($result, SORT_REGULAR);
```

It should be enough, if you can try

**Review by PierreRambaud — COMMENTED** [review]
Just one little comment

**Review by NeOMakinG — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
HI @Progi1984 

Thank you for your PR, I tested it and it works as we can see : 

https://user-images.githubusercontent.com/100699445/161931019-effd8bf6-5a98-4c6e-b9ca-5b3f53f4a181.mp4

PS : we have one little red information but I know that not because of your pr, moreover this message come only one time

**Comment by Progi1984** [comment]
Thanks @AureRita

**Pull request merged**